Tasks/NodeTool/nodetool.ts

```diff
@@ -27,149 +32,136 @@ async function run() {
+            // query nodejs.org for a matching version
+            version = await queryLatestMatch(versionSpec);
             if (!version) {
                 throw new Error(`Unable to find Node version '${versionSpec}' for platform ${osPlat} and architecture ${osArch}.`);
             }
 
-            toolLib.debug('version from index.json ' + version);
-            toolLib.debug('isLTS:' + ltsMap[version]);
+            // check cache
+            toolPath = toolLib.findLocalTool('node', version)
         }
 
-        //
-        // Download and Install
-        //
-        toolLib.debug('download ' + version);
-
-        // a tool installer intimately knows how to get that tools (and construct urls)
-        let fileName: string = osPlat == 'win32'? 'node-' + version + '-win-' + os.arch() :
-                                            'node-' + version + '-' + osPlat + '-' + os.arch();  
-        let urlFileName: string = osPlat == 'win32'? fileName + '.7z':
-                                                     fileName + '.tar.gz';  
-
-        let downloadUrl = 'https://nodejs.org/dist/' + version + '/' + urlFileName; 
-
-        let downloadPath: string = await toolLib.downloadTool(downloadUrl);
-
-        //
-        // Extract the tar and install it into the local tool cache
-        //
-        let extPath: string;
-        if (osPlat == 'win32') {
-            taskLib.assertAgent('2.115.0');
-            extPath = taskLib.getVariable('Agent.TempDirectory');
-            if (!extPath) {
-                throw new Error('Expected Agent.TempDirectory to be set');
-            }
-
-            extPath = path.join(extPath, 'n'); // use as short a path as possible due to nested node_modules folders
-            extPath = await toolLib.extract7z(downloadPath, extPath);
-        }
-        else {
-            extPath = await toolLib.extractTar(downloadPath);
+        if (!toolPath) {
+            // download, extract, cache
+            toolPath = await download(version);
         }
-
-        // node extracts with a root folder that matches the fileName downloaded
-        let toolRoot = path.join(extPath, fileName);
-        
-        toolLib.cacheDir(toolRoot, 'node', version);
     }
 
-    console.log('using version: ' + version);
-
     //
     // a tool installer initimately knows details about the layout of that tool
     // for example, node binary is in the bin folder after the extract on Mac/Linux.
     // layouts could change by version, by platform etc... but that's the tool installers job
     //
-    let toolPath: string = toolLib.findLocalTool('node', version);
     if (osPlat != 'win32') {
         toolPath = path.join(toolPath, 'bin');
     }
 
-    console.log('using tool path: ' + toolPath);
-
     //
     // prepend the tools path. instructs the agent to prepend for future tasks
     //
     toolLib.prependPath(toolPath);
-    console.log();
+}
+
+async function queryLatestMatch(versionSpec: string): Promise<string> {
```

Comment:
I moved this code into a separate function to improve the readability of the main function. Other than yanking out the LTS logic, this code is largely the same.